linera-core/src/unit_tests/client_tests.rs

```diff
@@ -1643,76 +1680,1188 @@ where
+        .inner()
+        .block()
+        .unwrap()
+        .operations
+        .contains(&Operation::System(SystemOperation::Transfer {
+            owner: None,
+            recipient: Recipient::Burn,
+            amount: Amount::from_tokens(1),
+            user_data: UserData::default(),
+        })));
+
+    // Block before that should be b0
+    assert!(hashed_certificate_values[1]
+        .inner()
+        .block()
+        .unwrap()
+        .operations
+        .contains(&Operation::System(SystemOperation::ReadBlob {
+            blob_id: blob0_id
+        })));
+    assert!(hashed_certificate_values[1]
+        .inner()
+        .block()
+        .unwrap()
+        .operations
+        .contains(&Operation::System(SystemOperation::PublishBlob {
+            blob_id: blob1_id
+        })));
+
+    Ok(())
+}
+
+#[test_case(MemoryStorageBuilder::default(); "memory")]
+#[cfg_attr(feature = "storage-service", test_case(ServiceStorageBuilder::new().await; "storage_service"))]
+#[cfg_attr(feature = "rocksdb", test_case(RocksDbStorageBuilder::new().await; "rocks_db"))]
+#[cfg_attr(feature = "dynamodb", test_case(DynamoDbStorageBuilder::default(); "dynamo_db"))]
+#[cfg_attr(feature = "scylladb", test_case(ScyllaDbStorageBuilder::default(); "scylla_db"))]
+#[test_log::test(tokio::test)]
+async fn test_blob2<B>(storage_builder: B) -> anyhow::Result<()>
+where
+    B: StorageBuilder,
+    ViewError: From<<B::Storage as Storage>::StoreError>,
+{
+    // Configure a chain with two regular and no super owners.
+    let mut builder = TestBuilder::new(storage_builder, 4, 0).await?;
+    let description1 = ChainDescription::Root(1);
+    let description2 = ChainDescription::Root(2);
+    let chain_id2 = ChainId::from(description2);
+    let client0 = builder
+        .add_initial_chain(description1, Amount::ZERO)
+        .await?;
+    let mut client1 = builder
+        .add_initial_chain(description2, Amount::from_tokens(10))
+        .await?;
+    let pub_key1 = client1.public_key().await.unwrap();
+    let key_pair = KeyPair::generate();
+    let pub_key2 = key_pair.public();
+    let owner_change_op = Operation::System(SystemOperation::ChangeOwnership {
+        super_owners: Vec::new(),
+        owners: vec![(pub_key1, 50), (pub_key2, 50)],
+        multi_leader_rounds: 10,
+        timeout_config: TimeoutConfig {
+            fast_round_duration: Some(TimeDelta::from_secs(5)),
+            ..TimeoutConfig::default()
+        },
+    });
+    client1
+        .execute_operation(owner_change_op.clone())
+        .await
+        .unwrap();
+    let client2 = builder
+        .make_client(
+            chain_id2,
+            key_pair,
+            client1.block_hash(),
+            BlockHeight::from(1),
+        )
+        .await?;
+
+    // Take one validator down
+    builder.set_fault_type([3], FaultType::Offline).await;
+
+    let blob0 = HashedBlob::test_blob("blob0");
+    let blob0_id = blob0.id();
+
+    // Publish blob on chain 1
```

Comment:
…which is owned by client 0. :sweat_smile:
Maybe they should be `client1`, `client2a` and `client2b` or something?